index.d.ts

```diff
@@ -0,0 +1,50 @@
+interface ModifierArgs {
+  positional: unknown[];
+  named: { [key: string]: unknown };
+}
+
+interface IModifier<Args extends ModifierArgs = ModifierArgs> {
+  args: Args;
+  element: Element | null;
+  isDestroying: boolean;
+  isDestroyed: boolean;
+  didReceiveArguments(): void;
+  didUpdateArguments(): void;
+  didInstall(): void;
+  willRemove(): void;
+  willDestroy(): void;
+}
+
+type Owner = unknown;
+
+declare module "ember-class-based-modifier" {
+  export default class Modifier<Args extends ModifierArgs = ModifierArgs>
+    implements IModifier<Args> {
+    args: Args;
+    element: Element | null;
+    isDestroying: boolean;
+    isDestroyed: boolean;
+    constructor(owner: Owner, args: Args);
+    didReceiveArguments(): void;
+    didUpdateArguments(): void;
+    didInstall(): void;
+    willRemove(): void;
+    willDestroy(): void;
+  }
+}
+
+declare module "ember-class-based-modifier/classic" {
+  import EmberObject from "@ember/object";
+
+  export default class Modifier extends EmberObject implements IModifier {
```

Comment:
Should this not have a generic too?

Reply:
No: as my comment on the original PE explained in a bit more detail, classic classes basically can’t work with generics effectively. Alas.

Reply:
Follow-up question here: is there ever a scenario where we would see users *wanting* to subclass the classic class using native classes? It technically *works*, but I think we'd all recommend strongly against it here, as there's just no reason *to* do that here?

Reply:
The one scenario is that you are subclassing from an addon-provided class and you are stuck with their base-class of choice. I think that's probably quite rare though so I'm not too worried about it. (Although it doesn't seem to cost us much to support it by adding the generic either.)